go-controller/pkg/cni/cniserver.go

```diff
@@ -72,6 +75,9 @@ func NewCNIServer(factory factory.NodeWatchFactory, kclient kubernetes.Interface
 	}
 
 	if util.IsNetworkSegmentationSupportEnabled() {
+		if !config.UnprivilegedMode {
+			s.nadController = nadController
+		}
```

Comment:
Not sure what this is really doing. My ignorance but a comment would be welcome.

Reply:
so in CNI we have 2 modes: unprivileged and privileged mode. In unprivileged mode, ovnkube-node doesn't have sys admin ability to be able to wire the interfaces. I can't remember if its also missing other capabilities. In this mode, the CNI shim (executed by kubelet) wires the interfaces vs. privileged mode, where cni shim sends a request to ovnkube cni server and it performs the wiring.

For UDN so far, we have only supported privileged mode:
https://github.com/openshift/ovn-kubernetes/blob/master/go-controller/pkg/cni/cni.go#L165

Technically I don't really need to check here for unprivileged mode...I was just trying to not add a nadController unless I needed to (in order to not waste resources on the system), but now that it is inherited from the node itself there is no cost. I'll just remove this conditional.